pytensor/tensor/optimize.py

```diff
@@ -0,0 +1,523 @@
+        )
+
+
+class MinimizeScalarOp(ScipyWrapperOp):
+    __props__ = ("method",)
+
+    def __init__(
+        self,
+        x: Variable,
+        *args: Variable,
+        objective: Variable,
+        method: str = "brent",
+        optimizer_kwargs: dict | None = None,
+    ):
+        self.fgraph = FunctionGraph([x, *args], [objective])
+
+        self.method = method
+        self.optimizer_kwargs = optimizer_kwargs if optimizer_kwargs is not None else {}
+        self._fn = None
+        self._fn_wrapped = None
+
+    def perform(self, node, inputs, outputs):
+        f = self.fn_wrapped
+        x0, *args = inputs
+
+        res = scipy_minimize_scalar(
+            fun=f,
+            args=tuple(args),
+            method=self.method,
+            **self.optimizer_kwargs,
+        )
+
+        f.clear_cache()
+        outputs[0][0] = np.array(res.x)
+        outputs[1][0] = np.bool_(res.success)
+
+    def L_op(self, inputs, outputs, output_grads):
+        x, *args = inputs
+        x_star, _ = outputs
+        output_grad, _ = output_grads
+
+        inner_x, *inner_args = self.fgraph.inputs
+        inner_fx = self.fgraph.outputs[0]
+
+        implicit_f = grad(inner_fx, inner_x)
+        df_dx = grad(implicit_f, inner_x)
+
+        df_dthetas = [
+            grad(implicit_f, arg, disconnected_inputs="ignore") for arg in inner_args
+        ]
+
+        replace = dict(zip(self.fgraph.inputs, (x_star, *args), strict=True))
+        df_dx_star, *df_dthetas_stars = graph_replace(
+            [df_dx, *df_dthetas], replace=replace
+        )
+
+        grad_wrt_args = [
+            (-df_dtheta_star / df_dx_star) * output_grad
+            for df_dtheta_star in df_dthetas_stars
+        ]
+
+        return [zeros_like(x), *grad_wrt_args]
+
+
+def minimize_scalar(
+    objective: TensorVariable,
+    x: TensorVariable,
+    method: str = "brent",
+    optimizer_kwargs: dict | None = None,
+):
+    """
+    Minimize a scalar objective function using scipy.optimize.minimize_scalar.
+    """
+
+    args = [
+        arg
+        for arg in graph_inputs([objective], [x])
+        if (arg is not x and not isinstance(arg, Constant))
+    ]
+
+    minimize_scalar_op = MinimizeScalarOp(
+        x,
+        *args,
+        objective=objective,
+        method=method,
+        optimizer_kwargs=optimizer_kwargs,
+    )
+
+    input_core_ndim = [var.ndim for var in minimize_scalar_op.inner_inputs]
+    input_signatures = [
+        f'({",".join(f"i{i}{n}" for n in range(ndim))})'
+        for i, ndim in enumerate(input_core_ndim)
+    ]
+
+    # Output dimensions are always the same as the first input (the initial values for the optimizer),
+    # then a scalar for the success flag
+    output_signatures = [input_signatures[0], "()"]
+
+    signature = f"{','.join(input_signatures)}->{','.join(output_signatures)}"
+    return Blockwise(minimize_scalar_op, signature=signature)(x, *args)
```

Comment:
I don't understand what you're blockwising here? Are you trying to vectorize the scalar optimization to batch x, objective? But how do you know what's batch or not in args? 

Anyway, it seems your Blockwise core case is exactly the same that ends up being called with x, args, so it's always useless?

Reply:
Indeed. I'll have to think more carefully about it. I guess it's not actually possible to automatically determine what the signature should be. 

I'll take it out and users can vectorize it themselves if they need it.